src/components/expression/expression-loader.js

```diff
@@ -24,101 +31,213 @@ genotet.utils.inherit(genotet.ExpressionLoader, genotet.ViewLoader);
- * @param {string} conditionRegex Regex for experiment condition selection.
- * @private
+ * @param {string} dataName Name of the expression matrix data file.
  */
-genotet.ExpressionLoader.prototype.loadExpressionMatrix_ = function(matrixName,
-    geneRegex, conditionRegex) {
-  this.signal('loadStart');
-  var params = {
-    type: 'expression',
-    matrixName: matrixName,
-    geneRegex: geneRegex,
-    conditionRegex: conditionRegex
+genotet.ExpressionLoader.prototype.loadExpressionMatrixInfo =
+  function(matrixName, dataName) {
+    var params = {
+      type: 'expression-info',
+      matrixName: matrixName
+    };
+
+    $.get(genotet.data.serverURL, params, function(data) {
+        // Store the last applied data selectors.
+        _.extend(data, {
+          matrixName: matrixName
+        });
+
+        if (Object.keys(data.allGeneNames).length == 0) {
+          genotet.warning('input gene not found');
+          return;
+        }
+        if (Object.keys(data.allConditionNames).length == 0) {
+          genotet.warning('input condition not found');
+          return;
+        }
+
+        this.data.matrixInfo = data;
+        this.signal('matrixInfoLoaded');
+      }.bind(this), 'jsonp')
+      .fail(this.fail.bind(this, 'cannot load expression matrix', params));
   };
 
-  $.get(genotet.data.serverURL, params, function(data) {
-      // Store the last applied data selectors.
-      _.extend(data, {
-        matrixname: matrixName,
-        geneRegex: geneRegex,
-        conditionRegex: conditionRegex
-      });
+/**
+ * Implements the expression matrix loading ajax call. Since the matrix may
+ * contain a large number of entries, we use GET request.
+ * @param {string} matrixName Name of the expression matrix.
+ * @param {string} dataName Name of the expression matrix data file.
+ * @param {!Array<string>} geneNames Names for gene selection.
+ * @param {!Array<string>} conditionNames Names for experiment condition
+ *      selection.
+ * @private
+ */
+genotet.ExpressionLoader.prototype.loadExpressionMatrix_ =
+  function(matrixName, dataName, geneNames, conditionNames) {
+    var params = {
+      type: 'expression',
+      matrixName: matrixName,
+      geneNames: geneNames,
+      conditionNames: conditionNames
+    };
+
+    $.get(genotet.data.serverURL, params, function(data) {
+        // Store the last applied data selectors.
+        _.extend(data, {
+          matrixName: matrixName,
+          dataName: 'b-subtilis'
+        });
 
-      this.data.matrix = data;
+        if (data.geneNames.length == 0) {
+          genotet.warning('input gene not found');
+          return;
+        }
+        if (data.conditionNames.length == 0) {
+          genotet.warning('input condition not found');
+          return;
+        }
 
-      this.signal('loadComplete');
-    }.bind(this), 'jsonp')
-    .fail(this.fail.bind(this, 'cannot load expression matrix', params));
-};
+        this.signal('loadStart');
```

Comment:
Incorrect placement of loadStart. You need to place loadStart before ajax call. Merge #55 and use wrapper.